src/pds/ingress/pds-nucleus-product-completion-checker.py

```diff
@@ -0,0 +1,321 @@
+"""
+=========================================
+pds-nucleus-product-completion-checker.py
+=========================================
+
+Lambda function to check if the staging S3 bucket has received a complete product 
+with all required file. This lambda function is triggered by S3 bucket events
+when a new file is copied to the staging S3 bucket.
+
+"""
+
+import json
+import urllib.parse
+import shutil
+import boto3
+import os
+from xml.dom import minidom
+from boto3.dynamodb.conditions import Key, Attr
+
+print('Loading pds-nucleus-product-completion-checker function')
+
+s3 = boto3.client('s3')
+dynamodb = boto3.resource('dynamodb')
+
+# Main lambda handler
+def lambda_handler(event, context):
+
+    # Get the object from the event and show its content type
+    bucket = event['Records'][0]['s3']['bucket']['name']
+    key = urllib.parse.unquote_plus(event['Records'][0]['s3']['object']['key'], encoding='utf-8')
+    s3_url_of_data_file = "s3://" + bucket + "/" + key
+    s3_url_of_product_label = "s3://" + bucket + "/" + key
+    efs_mount_path = os.environ.get('EFS_MOUNT_PATH')
+    print("Lambda Request ID:", context.aws_request_id)
+
+    try:
+
+        # Product label received
+        if s3_url_of_data_file.lower().endswith(".xml"):
+            
+            missing_files = handle_missing_files_for_product_label(s3_url_of_product_label, bucket, key)
+
+            copy_file_to_efs(bucket, key, efs_mount_path)
+
+            # If there are any missing files for the given product label
+            if (len(missing_files) > 0):
```

Comment:
PEP8 style: no need for `(…)` parentheses on this `if` expression

Reply:
@nutjob4life , Java habits :-). Changing it.